projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-transient.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * REST API endpoint for editing Jetpack Transients.
+ *
+ * @package automattic/jetpack
+ * @since 9.7.0
+ */
+
+/**
+ * Jetpack transients API.
+ *
+ * @since 9.7.0
+ */
+class WPCOM_REST_API_V2_Endpoint_Transient extends WP_REST_Controller {
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'wpcom/v2';
+		$this->rest_base = 'transients';
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	/**
+	 * Called automatically on `rest_api_init()`.
+	 */
+	public function register_routes() {
+		// DELETE /sites/<blog-id>/transients/$name route.
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<name>\w{1,172})',
+			array(
+				array(
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => array( $this, 'delete_transient' ),
+					'permission_callback' => array( $this, 'delete_transient_permissions_check' ),
+					'args'                => array(
+						'name' => array(
+							'description'       => __( 'The name of the transient to delete.', 'jetpack' ),
+							'required'          => true,
+							'type'              => 'string',
+							'sanitize_callback' => 'sanitize_text_field',
+						),
+					),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Delete transient callback.
+	 *
+	 * @param \WP_REST_Request $request Full details about the request.
+	 * @return array|WP_Error
+	 */
+	public function delete_transient( \WP_REST_Request $request ) {
+		$transient_name = $request->get_param( 'name' );
+		if ( false !== strpos( $transient_name, 'jetpack_connected_user_data_' ) &&
+			get_current_user_id() === (int) substr( $transient_name, 28 ) ) {
```

Comment:
Let's move this bit into `delete_transient_permissions_check()`, together with the error it returns. That way all permission checks are in the same place.

Reply:
Thanks! Moved here: https://github.com/Automattic/jetpack/pull/19632/commits/199e2b346e7079fb3b59b8ee93d3556900d9a41d